2nd edition revision

dansanderson opened the pull request:
Goals for this revision:

1. Focus on the MEGA65 production model. The 1st edition pre-dated the first production model and was written with early project participants in mind. Speculations about how the production model would behave have been corrected, and information that does not apply to the production model has been removed.
2. Focus on new users. 1st edition bounced between new-user material and obscure technical topics. Emphasis has been streamlined, including the removal of material irrelevant to new users. Topics are task-oriented and prioritize common user goals.
3. Update procedures and screenshots to represent the most recent best practices and v0.96 release behaviors.
4. Enhance information about upgrading the platform components, including how to register with Filehost and get the latest release package.
5. Extend information about using disks and disk images.

Still to do: a short chapter on Ethernet file transfer. This will necessarily involve introducing M65Connect and/or mega65-tools. This feels like an expansion of scope, but I'm eager to take the position that Ethernet file transfer will be a major mode of operation for many users. It's faster and more reliable than hassling with SD cards. M65Connect should support Ethernet file transfers by the time this goes to press.

In general, this draft pushes the frame of reference of the User's Guide closer to our everyday experience, including more references to Internet resources. It's difficult to balance describing the MEGA65 in isolation as if it were a Commodore product and describing it as a living project with an online community. The more reality we bring into the printed books, the faster that information will go stale on the page. We could alternatively consider setting up new mega65.org short links to wiki pages for some of these topics and direct readers to web resources instead of trying to capture them in the User Guide.

nobruinfo commented on using-disks.tex:
```diff
@@ -0,0 +1,357 @@
+  \includegraphics[width=0.7\linewidth]{images/floppy_hd.jpg}
+\end{center}
+
+\underline{Note}: Make sure that the tape is opaque, such as masking tape (not Scotch tape), and that it covers both sides of the hole.
```
Recent discussion on Discord showed that the following could be considered:

* The opaqueness of the used tape shouldn't be relevant as all known PC floppy disk drives use mechanical switches.
* The photo just above the line shows a tape covering the upper side of the disk. Usually the switch is reaching partly into the hole from the under side. Covering only the top in such cases is leading the drive to handle the inserted disk still as HD.
* Applying one single piece of tape to cover both holes usually makes the disk wider ever so slightly. Since the force of the eject mechanism is limited disk with such treatment tend to remain stuck. A pair of pliers will in some cases be needed to pull out the disk. It would be better to use two pieces of tape or a much less thicker one than the one being shown on the picture.

dansanderson replied:
Fixed the statements about opaqueness, thanks! I've also made sure the text emphasizes covering both sides of the hole. It sounds like a split view photo showing both sides would be useful. The important part is to clearly indicate the location of the hole to cover: an inexperienced user might cover the write-protect hole.

I've never heard of two pieces of tape needing to be used, and would be concerned about too small a piece with too little surface to grip on being more likely to come undone inside the drive. Just interacting with my own MEGA65 it seems like there's plenty of clearance for a single piece of masking tape. (I've heard some people suggest electrical tape which I definitely would not recommend due to thickness and lack of grip.)

(Style note: I feel weird including a photograph in the manual at all, and might consider replacing it with a line drawing if I or someone else can draw one. That would at least avoid a specific implication of the kind of tape to use.)

nobruinfo replied:
I didn't want to say two pieces are needed. My wording was bad and my intention was only to point out that *if* covering the upper hole had to be shown, covering the underside hole would be the effective action to take. :)

Unfortunately I'm no good in creating vector drawings with the here needed detail being a piece of tape applied to a floppy disk.

gurcei commented:
Will jot some general thoughts here, and after that, I'll drop my other notes inline with the change-diffs:
- parts of our manual preferred manually adding line-breaks at a certain line-width, and on other occasions it makes lengthy lines that only line-break at the end of a paragraph. Maybe the style-guide should specify one way or the other. Just find that when commits switch from one way to the other it adds a bit of noise in the diff. (for me, I lean towards manual line-breaks at a certain width, but maybe it's something we can discuss in the manuals channel in discord.

- For commit#7f421bf7717033db129f43dade3c109798e419de
  - cosmetic page breaks make me a little nervous, as while they likely look great in the present state, they might not look so great in future when new text is added, or text is moved/removed. (could lead to unsightly page-breaks with pages of mostly empty space).
  - I haven't had a chance to assess the intent behind these page-breaks yet. So just wondering if there are any other means to achieve what you set out to do?
  - were you trying to group a set of content together on a single page? if so, I think we've encountered this scenario before and might have alternate solutions for it (to group various elements together to assure that they stay on the same page)

gurcei commented:
I'll also aim for my 1st review to drop notes relating to main chapters. I'll try do this tonight.

Then on my 2nd review, I'll focus on your BASIC changes and drop those comments in.

gurcei commented:
A side-note, could be a good time to update "introduction.tex" to say:
- FROM: ...1080 pages and growing!
- TO: ...1200 pages and growing!

gurcei commented:
Also, I noticed in the user-guide's chapter 2 blue page, the "Switching the MEGA65 on for the First Time" heading gets broken into two lines, with the 2nd line jutting out awkwardly:

![](https://gurce.net/imgs/img_2023_09_12_4_37_57.png)

Interestingly, the same page in the mega65-book.pdf doesn't suffer from this problem (perhaps the font is smaller there and it fits on a single line?)

Not sure if there's anything we can do to remedy this to look a bit nicer? Maybe with some latex-fu?

dansanderson commented:
> * parts of our manual preferred manually adding line-breaks at a certain line-width, and on other occasions it makes lengthy lines that only line-break at the end of a paragraph. Maybe the style-guide should specify one way or the other. Just find that when commits switch from one way to the other it adds a bit of noise in the diff. (for me, I lean towards manual line-breaks at a certain width, but maybe it's something we can discuss in the manuals channel in discord.

I agree that ideally we'd pick a style, document it specifically, and be consistent. Hard wrap and soft wrap both have advantages, and I don't have a strong opinion. Something we can encode in .vscode/settings.json for the project would be ideal.

I went with soft wrap for the new material because with VSCode LaTeX Workshop I was spending too much time re-formatting paragraphs (Alt-Q) to match style with hard wraps, and it didn't seem worth it because our existing hard-wrapped content is wildly inconsistent. You have to touch every paragraph after every change because it won't do it automatically, presumably to avoid re-wrapping something it shouldn't. If there's a way to get it to behave fully automatically with hard wraps, I'll take it. Note that with soft wraps you do have to activate soft wraps each time you open the file (Ctrl-Z on a Mac), but so far that's been less hassle than reformatting each paragraph after a change.

>   * cosmetic page breaks make me a little nervous, as while they likely look great in the present state, they might not look so great in future when new text is added, or text is moved/removed. (could lead to unsightly page-breaks with pages of mostly empty space).
>   I haven't had a chance to assess the intent behind these page-breaks yet. So just wondering if there are any other means to achieve what you set out to do?
>   were you trying to group a set of content together on a single page? if so, I think we've encountered this scenario before and might have alternate solutions for it (to group various elements together to assure that they stay on the same page)

I was borrowing the current practice already in place throughout the 1st ed. We might be able to avoid some cases with `\nopagebreak`. I'm not sure that's always better than page breaks, in general. Grouping decisions may be appropriate for today's text and inappropriate for tomorrow's, just like the page breaks. I would expect to tweak layout—and tweak content to fit layout—whenever preparing text for print. (For example, I've already reorganized section headings to prevent the chapter intro page from spilling over to two pages, even if the original section layout made sense logically.)

> Also, I noticed in the user-guide's chapter 2 blue page, the "Switching the MEGA65 on for the First Time" heading gets broken into two lines, with the 2nd line jutting out awkwardly:
> 
> ![](https://camo.githubusercontent.com/56fe2928f567b20a5cf3aadb98759efc028b269d77a1e1044d92ccb393ad7bb3/68747470733a2f2f67757263652e6e65742f696d67732f696d675f323032335f30395f31325f345f33375f35372e706e67)
> 
> Interestingly, the same page in the mega65-book.pdf doesn't suffer from this problem (perhaps the font is smaller there and it fits on a single line?)
> 
> Not sure if there's anything we can do to remedy this to look a bit nicer? Maybe with some latex-fu?

I thought we had modeled this specifically after Commodore's C64 books, including this indentation style on the chapter intro page. I'm looking at the PDFs now and am surprised to see that their chapter intro pages use much smaller text. Maybe I'm looking at the wrong editions? I'm not personally bothered by this example but I don't object to seeing it changed.

gurcei commented on using-disks.tex:
```diff
@@ -0,0 +1,368 @@
+
+The MEGA65 also includes a ``virtual'' disk drive that can mount D81 disk image files\index{D81 disk image} stored on the SD card. Most MEGA65 software that you download from the Internet is in the form of a D81 disk image. You can create a new D81 disk image directly from the MEGA65, and start saving your BASIC programs to the SD card without any additional hardware. You can also copy files betw ...
+
+The Intro Disk Menu\index{Intro Disk} that you saw when you first switched on the computer is a program on a D81 disk image, a file named {\tt MEGA65.D81} on the SD card. The MEGA65 is initially configured to boot this disk image automatically. You can change this in the Configuration Utility.\index{Configuration!Utility} (Refer back to chatper \vref{cha:configuringyourmega}.)
```
typo: chatper / chapter

gurcei commented on using-disks.tex:
```diff
@@ -0,0 +1,368 @@
+\index{Disk Drives!Terminology}
+\index{Connections!IEC}
+
+Each disk drive (physical or virtual) is accessed via a {\it unit} number.\index{Unit number (IEC devices)} With vintage Commodore computers, the unit number refers to an IEC device connected to the computer. Commodore reserved unit numbers in the range 0 -- 31 for devices of various purposes, with 8 -- 11 reserved for disk drives. If you've ever used a Commodore 64 and typed \screentext{LOAD "*" ...
```
> ... connects as unit 8

should be: "...**connected** as unit 8".

gurcei commented on using-disks.tex:
```diff
@@ -0,0 +1,368 @@
+
+With the MEGA65, you can assign a unit number to the virtual disk drive with a D81 disk image mounted, or to the internal 3.5" floppy drive. You must mount a disk image or the internal 3.5" floppy drive to a unit number before it can be used. Any message sent to a unit number assigned to a virtual disk or the internal floppy drive is handled by the MEGA65. All other messages are sent to the IEC s ...
+
+Disk commands also accept an optional parameter to specify a {\it drive} number.\index{Drive number (IEC devices)} This is only needed when connecting a vintage dual floppy drive via the IEC port, such as the Commodore 4040, 8050, or 8250. Every disk drive assigns drive number 0 to the first drive. Dual-drive units assign a drive number of 1 to the second drive. Dual disk drives are usually equip ...
```
> ... use device 0 by default

Should this say **drive 0** instead of 'device 0'?

I admit I'm not entirely clear on the distinction between drive/device and we might've had past discussions in discord on this topic

dansanderson replied:
Correct, this should be "drive." "device" is ambiguous. (For a while I was confused that "drive" referred to the 0/1 that selected the internal managed drive in the Freezer menu. I had to dig around to realize that this only applies to vintage dual drives. — I just bought a vintage dual drive last weekend! I doubt it'll work with the MEGA65 due to IEC timing issues but it'll be fun to try it in GO64 mode.)

gurcei commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
+\section{Upgrading the Factory Core in Slot 0}
+\index{Core!Upgrading Slot 0}
+
+It is possible to upgrade the factory-installed MEGA65 core in slot 0. You only need to do this in rare cases, such as if a newer version of the MEGA65 core includes changes or bug fixes for the start-up process. The process is elaborate, delicate, and could result in a MEGA65 that fails to start if something goes wrong. It is {\em strongly} recommended that you do not upgrade slot 0 unless the a ...
```
> ...fully functional in slot 1, without upgrading slot 0.

How about: "...without **needing to** upgrade slot 0".

gurcei commented on gettingstarted.tex:
```diff
@@ -114,35 +111,216 @@ \subsubsection{MEGA KEY}
+
+\section{Running Commodore 64 Software}
+
+The MEGA65 is capable of running Commodore 64 software. There are two ways to do this: the built-in GO64 mode,and the {\it C64 for MEGA65} FPGA core.
```
> GO64 mode,and the...

Add a space between the ",and"

gurcei reviewed (changes requested):
Ok, I've dropped in all my notes from my review of chapters 1 to 6.

Apologies for the number of notes, most of them are minor things that I hope won't be too taxing. Some points can just be a chance to hear out your perspective on things, while other points are me making a note of culled content we could potentially find a home for in future.

After this, I'll review Ch7 and basic chapter

dansanderson commented on settingup.tex:
```diff
@@ -69,68 +77,107 @@ \section{MEGA65 Screen and Peripherals}
+
+PAL and NTSC are analog video signal formats that affect the the resolution and vertical sync speed of the video output, even when using a modern digital display. Your display may support either mode, or it may only support one or the other. You can use this screen to test the modes with your display.
+
+Select and test you video configuration. For example, press \specialkey{TAB} to switch to the ``PAL 50HZ'' mode.
```
PAL 50HZ is quoting the user interface in this case, which is all caps. I changed this from quoted to boldface, which is used for UI text elsewhere.

dansanderson commented:
Thank you so much for the detailed review, @gurcei ! I have incorporated all of these.

gurcei commented:
No problem :-) Also, feel like some of the comments got hidden by github within a textbox saying "45 hidden conversations" (somewhere between the visible conversation above). Just thought I'd mention it, in-case they were overlooked. They expand once you click on that textbox a few times (they seem to expand by 20 per click of the textbox)

dansanderson commented:
> No problem :-) Also, feel like some of the comments got hidden by github within a textbox saying "45 hidden conversations" (somewhere between the visible conversation above). Just thought I'd mention it, in-case they were overlooked. They expand once you click on that textbox a few times (they seem to expand by 20 per click of the textbox)

Oh dang, I thought it was just hiding previously resolved comments. I'll do these now. 😅

dansanderson commented on gettingstarted.tex:
```diff
@@ -70,17 +68,14 @@ \subsubsection{THE CURSOR KEYS}
 
 The cursor moves in the direction indicated on the keys: \megakey{$\leftarrow$} \megakey{$\uparrow$} \megakey{$\rightarrow$} \megakey{$\downarrow$}.
 
-However, it is also possible to move the cursor up by using \specialkey{SHIFT} and \megakey{$\downarrow$}. In the same way you can move the cursor left by using \specialkey{SHIFT} and \megakey{$\rightarrow$}.
```
I don't like it, feels obscure. But if you like it I'll put it back. :)

dansanderson commented on gettingstarted.tex:
```diff
@@ -114,35 +111,216 @@ \subsubsection{MEGA KEY}
+\end{itemize}
+
+\needspace{4cm}
+You can create fun pictures just by using these colours and letters. Here's an example of what a student drew:
```
"year four" is a British-ism that not everyone understands. 4th year of what? I actually do not know in this case. I'm guessing it's the equivalent of the American "4th grade student," around age 10, but I don't know who actually drew this. For all I know this was a senior in college who drew this.

"Primary school student" maybe?

dansanderson replied:
I'll go with "4th year student" and just trust that most people know what it means, or that it doesn't matter. :)

dansanderson commented on screen-editor.tex:
```diff
@@ -1,143 +0,0 @@
-
-You will now be able to type those additional ASCII characters via the keyboard. To revert back to the original PETSCII character set, type \screentext{FONT C}.
-
-\textbf{Extras}
-
-Here are some extra, useful things you can do whilst in the screen editor:
-
-To enter the \textbf{Freezer}, press and hold \widekey{RESTORE} for half to one second, then press \megakey{J} to
-switch controller ports without having to physically swap the controller to the other port. You can read more about
-the Freezer, and what else it can do on page \pageref{sec:freezer}.
-
-\index{BASIC 65 Examples!POKE}
-  To enter \textbf{Fast mode} (40.5MHz), either type \screentext{SPEED} (preferred way), or \screentext{POKE 0,65} or enter the Freeze Menu and switch the CPU frequency with \megakey{F}.
-
-  Return back to \textbf{Slow mode} (1MHz) by either typing \screentext{SPEED 1} (preferred way), or \screentext{POKE 0,64} or enter the Freeze Menu and switch the CPU frequency with \megakey{F}.
-
-  \megasymbolkey + \specialkey{SHIFT} switches between uppercase and lowercase text for the entire display.
```
The only thing in this section that has anything to do with the screen editor is case switching, and that now has a prominent mention in another section in this chapter. CPU speed, the POKE and SPEED commands, and the Freezer are unrelated to the screen editor. The SPEED command is in the BASIC reference, and the Freezer is covered in detail elsewhere in the User's Guide.

The general idea is that a MEGA65 user will mostly know the MEGA65 mode of the machine, and that makes CPU speed an advanced topic. If we're going to teach a new user that they can underclock the CPU, there should be something the user wants to accomplish that requires it. I've seen people use it to slow down program and directory listings, but that's all I can think of.

These POKEs don't work from MEGA65 mode as far as I can tell. I still wouldn't include them here even if they did, but maybe that's something else we can look into.

dansanderson replied:
FWIW, I found other mentions of POKE 0,65 in the Book. I'm still confident that they don't work from MEGA65 mode. The actual FAST register is D031.6, and this changes with the SPEED command, but not with POKE 0,65. 🤷‍♂️

dansanderson commented on gettingstarted.tex:
```diff
@@ -114,35 +111,216 @@ \subsubsection{MEGA KEY}
+To open the Freezer menu, hold the \widekey{RESTORE} key for one second, then release it. The MEGA65 will pause whatever it is doing, flicker the border color, then open the Freezer menu. Whatever program was running remains in memory and can be resumed by pressing the \specialkey{F3} key. You can also abandon the running program and reset the MEGA65 by pressing \specialkey{F5}.
+
+\begin{center}
+  \includegraphics[width=0.7\linewidth]{images/freezer.png}
```
Fixed in the reshoot using m65 -S. (I still need to follow up on the suggestion in Discord that I might have the wrong version of the Freezer installed, which is possible.)

dansanderson commented on gettingstarted.tex:
```diff
@@ -114,35 +111,216 @@ \subsubsection{MEGA KEY}
+
+GO64 mode is actually just a temporary re-configuration of the MEGA65. All of the MEGA65's features are still present, including the Freezer menu for mounting D81 disk images.
+
+Much Commodore 64 software can be found on the Internet in the form of D64 disk images. The MEGA65 only supports D81 disk images via the SD card and Freezer menu. You can use a peripheral such as the SD2IEC\index{SD2IEC device} with the MEGA65's IEC port to use D64 disk images. Be sure to obtain an SD2IEC with an independent power supply, and not one that depends on a Commodore 64 tape connector. ...
```
This was intending to address D64 disk images specifically, immediately following the clarification that in GO64 mode the only virtual disk option is D81 disk images. This also prepares for the mention of D64 image support in the C64 core in the subsequent section. (IMO this paragraph *is* the short version pointing to another location for details.)

We could add a mention of IEC device support here. It's a top-level section dedicated to running C64 software, I think it'd be a good fit.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
-\label{sec:configuration-utility}
+You can use the microSD memory card slot on the rear of the MEGA65 as persistent storage for the computer's configuration and system files. Having a prepared card in this slot overrides the SD card installed inside the computer. Having a microSD card installed is convenient if you wish to move it between your MEGA65 and your PC.
 
-The configuration utility for the MEGA65 has a similar purpose to the BIOS on a PC, which is to allow you to control certain default behaviours of your MEGA65. However, rather than storing the configuration data in
-battery-backed RAM, the MEGA65 stores this data on sector 1 of the SD card. If you switch SD cards, you will change the configuration data.
```
The fact that config is stored on the SD card is in the rewrite, as is the fact that it's in a hidden partition. "Sector 1" is not useful information to users. If you just mean that it should be available to someone who might need that information (people developing the Config utility?) then sure, just not the User's Guide.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
 
-This will present the MEGA65 Utility Menu, which contains a
-selection of built-in utilities, similar to the following:
+To access the Configuration Utility, switch off the MEGA65, hold the \megakey{ALT} key and switch it back on. The Utility menu\index{Utility menu} appears with several options. Press \megakey{1} to start the Configuration Utility.
```
Fixed. Do point all of these out. I tried to use gettingstarted.tex as a reference for how to draw each of these keys, and I did not intend to change anything, I probably just got some wrong.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
 
-%\begin{minipage}{\linewidth}
-  If your MEGA65's System Partition is corrupted, you may be
-  prompted to press \megakey{F14} to correct this, (i.e., hold \specialkey{SHIFT} and press
-  \megakey{F13}), with a display similar to the following:
+Formatting an SD card erases its contents, and this operation cannot be undone. We recommend that you do not erase the internal SD card that came with the computer.
 
-%  \vspace{5mm}
-\begin{center}
-  \includegraphics[width=0.7\linewidth]{images/ss-m65config-corrupt.png}
-\end{center}
-%\end{minipage}
-
-To correct this error, press \megakey{F14}. Next, press \megakey{F7} to save the reset configuration, otherwise the reset data will not be saved to the MEGA65 System
-Partition.
```
The instructions are printed in the error message. I'm not sure this edge case needs explicit documentation.

dansanderson replied:
fwiw I did not delete the screenshot of the error message and will leave that in the files: images/ss-m65config-corrupt.png.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
-    you use \megakey{$\leftarrow$} and \megakey{$\rightarrow$} to select another page and
-    return to this page, the values will update if a Real-Time Clock
-    is fitted and functioning.
-  \item \screentext{DMAGIC REVISION} allows selecting the default mode of
-    operation for the C65 DMAgic DMA controller.  This option is only
-    required for ROMs not detected by the MEGA65's HYPPO Hypervisor.
-    If you see screen corruption in BASIC,
-    try toggling this option.
-  \item \screentext{F011 DISK CONTROLLER}
-    \index{Disk Drives!D81 Images}
-    This option allows you to select whether the internal 3.5'' floppy
-    drive functions using real diskettes, or whether it simply makes
-    noises to add atmosphere when using D81 disk images from the SD
-    card.  This merely sets the default option, and you can change
-    this setting, or select a different disk image for use as either
-    or both of the C65 3.5'' DOS based drives.
-  \item \screentext{DEFAULT DISK IMAGE} allows you to choose the D81 disk image
-    used with the internal drive, if the \screentext{F011 DISK CONTROLLER}
-    option above is set to \screentext{USES SDCARD DISK IMAGE}. You can read more about
-    D81 disk images on page \pageref{sec:d81-images}.
-  \item \screentext{LONG FN SUPPORT} is a feature that is still under development
-    at time of writing, and we suggest leaving it disabled for now until the feature
-    matures in future bitstreams. Its aim is to provide long filename support for the
-    SD card.
```
See newer draft, I came to the same conclusion independently and fixed it. 👍

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
+\index{SD Cards}
 
-Press \specialkey{RETURN} to complete the configuration:
+Your MEGA65 is equipped with two SD card slots: a full-size SD card slot inside the case, and a microSD card slot accessible from the rear of the computer.\index{SD Cards!Locations} The MEGA65 uses the SD card for storing configruation settings, loading the operating system, updating the firmware, and storing your software and data as virtual disk images.
```
To be honest, I removed these originally because I don't believe the trap door is actually a good way to access the internal SD card. The design had good intentions but the positions didn't line up in the final product and it's quite difficult to remove the SD card this way.

Based on your suggestion I tried messing with the trap door again and decided I could go either way, so I have added them back.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
-  Next, press the number corresponding to the \screentext{CONFIGURE MEGA65} item.  The MEGA65
-  Configuration Utility will launch, showing a display similar to
-  the following:
+An SD card formatted by another computer will {\em not} work with the MEGA65, even if it only erases the FAT32 partition. You {\em must} use the MEGA65 SD Card Utility to format the card.
```
It's what I would like to call it, yes. I don't want to have to explain what "FDISK+FORMAT" means. IMO the utility menu should use better names. I agree that this is in tension with what actually appears on the screen but I think the obscurity of the name justifies referring to it as a concept independent of the screen text.

FYI: There's a more subtle version of the "should we match the screen text?" question with regards to the "Freezer menu." It's called the "Freeze menu" on screen, but the books consistently refer to it as the "Freezer menu" so I kept it that way. (I called in the "Freeze menu" in the Welcome Guide to match the screen text.)

dansanderson replied:
I added the on-screen title to the menu description: `From the menu, select option \megakey{2} to start the SD Card Utility (\texttt{SDCARD FDISK+FORMAT UTILITY}).`

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
-always attempt to boot from the external microSD card if one is
-present.
-
-For safety, when formatting we {\em strongly} recommend
-that you remove any SD card or microSD card that you do not intend to
-format, so that you do not accidentally destroy any data.  This is
-because formatting an SD card on the MEGA65 cannot be undone, and
-all data currently on the SD card {\em will be lost}.  If you
-have files or data on the SD card that you wish to retain, you
-should first back them up.  The contents of the FAT32
-partition can be easily backed up by inserting the SD card into
-another computer.  The contents of the MEGA65 System Partition,
-including the contents of freeze slots requires the use of specialised
-software.
-
-You should aim to back up valuable data from your
-MEGA65 on a regular basis, especially while the computer remains under
-development.  While we take every care to avoid data corruption or
-other mishaps, we cannot guarantee that the MEGA65 is free of bugs in
-this regard.
```
I feel like I captured the warnings well enough. It's debatable whether the project is still in an "our bugs could delete all your data" phase. :) Interesting point about the system partition, though the previous text isn't helpful by referring to "specialised software" without a hint.

dansanderson replied:
Added this footnote at an appropriate place: `If you wish to make a backup of the complete SD card including the hidden partition, you must use a disk utility that copies entire partitions, not just the files on the FAT32 partition.`

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
 
-\subsection{Video}
-\label{configuring-video}
+The utility erases the SD card and sets up the partitions. When it is finished, it prompts to install the system files embedded in the MEGA65 chipset core. If you have installed an updated MEGA65 core in slot 1, select it, otherwise select the factory-installed MEGA65 core in slot 0. (If you just received your MEGA65, slot 0 is the only option.)
```
Agreed. Maybe this here is the right place to explain it further. Will think some more.

This draft is mostly just trying to accommodate an explanation of the user experience. They get prompted to pick a core slot for embedded files even if there's only one MEGA65 core installed. And there's the edge case of picking a core that's missing the ROM, which I felt compelled to put in a footnote but maybe User's Guide readers won't actually encounter that one. (I see it all the time but I run dev cores. :) )

dansanderson replied:
I did some modest rewriting of a single paragraph to make this less bewildering on first read. It's not a deep explanation but it's better than just dropping the concept as if the user is supposed to know what it is.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
+The MEGA65 supports the Commodore 1351 mouse,\index{Commodore 1351 mouse} the Commodore Amiga mouse,\index{Commodore Amiga mouse} or modern equivalents such as a USB mouse connected with a \href{https://retrohax.net/shop/amiga/mouster/}{mouSTer} adapter.\index{mouSTer adapter} The port must be set to the correct mouse type, where {\bf normal} refers to the 1351 mouse. If an Amiga mouse is connect ...
 
-It is also possible to attempt a recovery from a lost Master Boot Record error (also known as the Boot Sector), by typing \screentext{FIX MBR} instead.
+{\bf Amiga mouse detection} controls how the MEGA65 interprets Amiga mouse signals and converts them to equivalent 1351 signals when the port is in Amiga mouse mode. Some joysticks experience interference on a port with Amiga mouse mode enabled. If you notice any such interference, change this setting to {\bf conservative,} otherwise leave it set to {\bf aggressive.}
 
-The aim here is to have a correctly formatted SD card with all of the essential files stored on it so the MEGA65 can properly boot.
-When switching on, the MEGA65 will search for, and boot using these files:
-\begin{itemize}
-\item {\tt FREEZER.M65} (Freeze Menu program)
-\item {\tt AUDIOMIX.M65} (Freeze Menu audio mixer utility)
-\item {\tt C64THUMB.M65} (C64 thumbnail image used in freezer)
-\item {\tt C65THUMB.M65} (C65 thumbnail image used in freezer)
-\item {\tt MEGA65.ROM}   (128KB ROM file)
-\item {\tt MEGA65.D81} (Disk image)
-\end{itemize}
```
This feels like another case where the kind of user that needs this information would already have it, and wouldn't need to get it from the User's Guide. That said, I do like the idea of a beginner-level explainer of "what are all these weirdo files on the SD card that I can see from my PC?" so maybe this is worth restoring in this section...

dansanderson replied:
Added brief explanations and parentheticals about .D81, .M65, and .ROM files on the SD card. I'm not listing them, but hopefully this is useful information in the right places.

gurcei replied:
Just playing catchup now, just wanted to locate which point were the brief explanations added? (I missed out on examining the commit after the squashed-merged went through :D)

dansanderson replied:
Sorry, didn't mean to jump ahead of your feedback with the merge! I had assumed we could continue the review after the submit.

Configuring: "The system files (with filenames ending in {\tt .M65} and {\tt .ROM}) are embedded in the core..." MEGA65.D81 gets several prominent mentions. MEGA65.ROM is in a newer footnote. I forget all of the changes I made in direct response to this feedback.

I don't believe we should itemize the .M65 files in the User's Guide. At least for the posture of the User's Guide, the only reason to replace an individual file is if you're making a change to system software or helping someone test a change to system software.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
+If you no longer have the bundled software files, you can obtain them from the MEGA65 Filehost website. Visit the following URL, then search for ``MEGA65 Release SD Card - Intro Disk Extras.''
 
-% 2021-03-17 edits by SBC
+\url{https://files.mega65.org}
```
Maybe! That'd work in this case, but not as well in other cases where the documentation instructs the reader to find files on Filehost. Filehost does not (yet?) allow aggregating software for Windows/Mac/Linux under a single detail page, so later instructions tell the user to search for a title instead of offering three short links.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
-character set at \$D000.
+If {\bf F011 disk controller} is set to {\bf 3.5" floppy drive,} the boot process will pause just before the \screentext{READY.} prompt to check if a boot disk is inserted in the drive. If you do not use a physical boot disk, you may wish to leave this set to {\bf SDCARD disk image} for a faster boot process.
 
-You can optionally name your alternate ROMs as {\tt MEGA65x.ROM}, replacing {\tt x} with a number from 0 to 9.
-This allows you to quickly boot-up to your alternate ROMs by holding down a number from \megakey{0} to \megakey{9} prior
-to switching on your MEGA65.
```
One reason to put this back (here or elsewhere in User's Guide) is we bundle some extra ROMs on the SD card, and it'd be cool to direct new users to the fact that they can hold down the 1 key and see what a real C65 looked like.

dansanderson replied:
Added a new short section about booting alternate ROMs at the end of the Upgrading chapter, with an explanation as to why you might want to do this.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
-If you have sourced your own {\tt 911001.bin} C65 ROM and would like to patch it to the latest MEGA65 closed ROM,
-we do provide patches, as the additional improvements we have made to the closed rom are open source.
-These diff files are available at \url{http://mega65.org/rom-diffs}
+The {\bf Video} page configures video settings. These are the same settings from the on-boarding configuration, including the PAL or NTSC video mode, Digital Video sound, and CRT emulation.\index{Configuration!On-boarding}\index{Display!Setting PAL/NTSC}
 
-\textbf{MEGA65 Open ROMs}
+\begin{center}
+  \includegraphics[width=0.7\linewidth]{images/ss-m65config-3.png}
+\end{center}
 
-Another available option is to make use of \textbf{MEGA65 Open ROMs}. The latest version of this is always downloadable from either of the following urls:
+{\bf Video mode} selects between the PAL compatibility mode and the NTSC compatibility mode. You can also change this while running programs using the Freezer menu.
 
-\begin{itemize}
-  \item \url{http://mega65.org/open-roms}
-  \item \url{https://github.com/MEGA65/open-roms/raw/master/bin/mega65.rom}
-\end{itemize}
```
Yeah, really just trying to streamline here. I can see an argument for the User's Guide introducing new users to all of the major aspects of the MEGA65 project. Here, I went with "a MEGA65 user wants to use the MEGA65 platform" as a guiding principle. There's also an argument that this wouldn't get mentioned elsewhere, so it should be mentioned here.

Someone needs to bolster my confidence that the OpenROM is still a viable sub-project. My understanding is that it's no longer being worked on, is incomplete, and doesn't run most software that has been produced for "the MEGA65." I can understand propping it up as a statement of the open source values of the project, even if it's unfinished and unlikely to be finished.

dansanderson replied:
Also resolved by new Alternate ROMs section.

dansanderson commented on mega65-userguide.tex:
```diff
@@ -141,11 +141,10 @@
 \input{gettingstarted}
-\input{screen-editor}
 \input{configuring}
-\input{modes}
```
The good news with this one is that the LaTeX file is still in the repo and still included in the Compendium for now. :)

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
+
+\section{Obtaining the Latest Files}
+
+You can download the latest MEGA65 core, ROM, and system software from the MEGA65 Filehost website.\index{Filehost website} Due to distribution restrictions for the Commodore 65 ROM code, some files require a Filehost account registered to a MEGA65 owner to access. All owners of the MEGA65 have a license to all versions of this ROM code.\footnote{There is a procedure for non-owners to get the lat ...
```
This is a good one for a short URL. I send people to it often!

dansanderson replied:
This is now https://mega65.org/rom-faq, and it's live.

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
-    \item{You can also find .bit, .mcs and .prm files located here too.}
-  \end{itemize}
+\begin{enumerate}
+  \item Visit \href{https://files.mega65.org}{Filehost}. Click ``Sign Up.'' Follow the prompts to create an account.
```
Just above this the URL gets a formal introduction and appears on its own line. I just linked it to be thorough. But yes I will also change this to "the Filehost website."

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
+\begin{enumerate}
+  \item Click the ``Files'' tab of the Filehost website.
+  \item In the search box on the left-hand side, type: ``release'' The list will update to show only files with that word in the title.
+  \item Locate the entry named, ``MEGA65 Core Release Package (mega65r3) incl. ROM,'' where ``mega65r3'' matches your hardware revision.
```
Agreed, both done.

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
+  \item Click the entry. Confirm that this release package is for your hardware revision, then click ``Download'' to download the file.
+\end{enumerate}
+
+If you don't see an entry that says ``incl. ROM,'' check that you are signed in and that you have redeemed a valid owner code. Note that there is an entry for the Release Package that does not include the ROM that is visible to everyone. To ensure you are using a compatible set of files, get the package that says ``incl. ROM.''
```
Disagree. For typable strings yes (and so also I wouldn't present typable strings inline with quote marks), but in this case it's a quoted phrase seen on a website, so for typographical reasons the ending punctuation goes inside the quotes. It's not ambiguous.

(I realize most people prefer "logical" punctuation to typographical punctuation these days, but IMO it's especially important for print typography.)

gurcei replied:
Hmm, on this one, maybe I'll ask @Impakt's thought on how he feels about this.

For me, it was a little jarring, as for technical matters, if I see things in quotes, my eyes will look for those specific things within the quotes.

I had a read online on this topic:
- https://www.thesaurus.com/e/grammar/does-punctuation-go-inside-or-outside-quotation-marks/

A lot of the quote examples I see there seem to be of the spoken form, a quote of conversation of a character within a story, wanting to convey the slight pauses of that character's conversation.

So in that sense, yep, I believe I've seen punctuation used in that manner, though I don't feel like I've seen this pattern used in technical documentation.

dansanderson replied:
I agree that technical documentation should avoid typeable strings in regular quotes for this reason. Putting sentence punctuation outside of typographic quote marks is not a professional solution. The solution is to use a convention not based on quote marks. Casual and business comms are slipping towards `"blah",` but typesetters are not, because of the space below the quote mark and space above the sentence punctuation.

I used quotes in this case because it's not strictly a typeable string. It's English text, and it's not ambiguous. Nobody is going to see `"incl. ROM,"` in the text and `incl. ROM` on the website and think they are different. The alternative would be `{\bf incl. ROM}`. If it's really that important that this be considered a typeable string, then I can switch to that.

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
 
 \begin{center}
-\includegraphics[trim= 0  0 0 10mm,clip,width=0.7\linewidth]{images/ss-flashmenu.png}
+  \includegraphics[width=0.7\linewidth]{images/ss-flashmenu.png}
```
This would be good polish. I'm a bit exhausted from re-staging my only MEGA65 to look like a factory device, and 1st edition wasn't as careful as I've been. 😅 I'll leave this open in case I feel like deleting all my cores and taking another screenshot.

dansanderson replied:
ok done :)

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
-cannot be booted from:
+You can select a core to boot using the cursor keys and \specialkey{RETURN}, or you can simply press the number key that corresponds to the slot. The boot process continues with the new core. The MEGA65 will keep running the new core until you physically power it off. (Pressing the reset button will not reset which core is being run.)
+
+When you switch on the computer without opening the core selection menu, the MEGA65 looks for a core in slot 1. If there is a valid core in that slot, it uses it. Otherwise it tries slot 0.\footnote{You can change the default core slot from 1 to 2 by moving DIP switch \#4 to the ``on'' position. DIP switches are located inside the case, on the main board. For a diagram of the DIP switch locations ...
```
I believe the new feature is that any given core can be configured to be the auto-booting core for a given cartridge type. It is unrelated to the default slot in the non-cartridge case. When the feature is finished we can revisit this text—and maybe the new configuration feature needs to be described in this section of the User's Guide (maybe at the end).

dansanderson replied:
Added todo in source locations where this will go

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
+
+\newpage
+
+\section{Upgrading the Factory Core in Slot 0}
```
+1. This should all get another pass once this feature is finished. I believe these are the final intended instructions for v0.96 but we'll need to test them again.

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -1,311 +1,254 @@
+\item The Flash Menu then checks if \specialkey{NO\\SCROLL} is being held down.
 \item If it is, the Flash Menu program shows its display, allowing you to select or re-flash a core.
-\item If \specialkey{NO SCROLL} is \underline{not} being held down, the Flash Menu program checks if Flash Slot 1 contains a valid
+\item If \specialkey{NO\\SCROLL} is {\em not} being held down, the Flash Menu program checks if Flash Slot 1 contains a valid
```
Yeah, mixed feelings. We overload both underline and bold for various conventions, and emphasis is supposed to be for emphasis. But it's also the case that our choice of font and line spacing make the emphasis style appear pretty weak. Maybe this has to be a global replace of `\em` for `\underline`.

dansanderson commented on cores-and-flashing.tex:
```diff
@@ -315,44 +258,4 @@ \section{Understanding The Core Booting Process}
-memory, it will start the MEGA65 Factory Core from slot 0 again.
-
-\phantomsection
-\section{DIP Switches}
-\label{sec:dip-switches}
-
-The MEGA65 motherboard has 4 DIP switches (at position S3 on the PCB),
-which are documented in the following table.
-
-\begin{center}
-  \begin{tabular}{|>{\centering\arraybackslash}p{1.5cm}|p{8cm}|>{\centering\arraybackslash}p{1.5cm}|}
-    \hline
-    {\textbf{Switch\#}} & {\textbf{Purpose}} & {\textbf{Default}}\\
-    \hline
-    1 & {
-      Off = Disable Keyboard Flash \newline
-      On = Enable Keyboard Flash
-    } & Off \\
-    \hline
-    2 & {
-        Off = Disable remote control via Ethernet \newline
-    } & Off = Enable remote control via Ethernet \\
-    \hline
-    3 & {
-         Off = Disallow non-hypervisor access to QSPI flash \newline
-         On = Permit non-hypervisor access to QSPI flash
-    } & Off \\
-    \hline
-    4 & {
-         Off = Boot from slot 1 \newline
-         On = Boot from slot 2
-    } & Off \\
-    \hline
-  \end{tabular}
-\end{center}
```
Agreed it should be somewhere. At least two of these switches aren't useful to a typical user. I'm used to vintage computer books having stuff like this on one-page appendices, maybe that's justified in this case.

(I was impressed that on a RetroCombs live stream, DIP 4 was mentioned and Stephen seemed excited to know that  you can change the default core slot. I can't think of a reason why anyone would do that that isn't keeping their case open and testing multiple cores, but if someone else got excited about it then that's a data point in its favor. :) )

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
 \begin{center}
-\includegraphics[trim=0 8cm 0 0,clip,width=0.7\linewidth]{images/mega65_OpenROM_boot_noSD.png}
+  \includegraphics[width=0.7\linewidth]{images/ss-m65config-2.png}
 \end{center}
```
Do you know how to reproduce an openrom default boot? If I nix mega65.rom I just get "could not find rom mega65.rom" in hyppo and it loops. Same if I nix all system files. Is this still a feature?

gurcei replied:
Yeah, this is a funny one. For a long time, the way the docs described it wasn't the way it actually triggered.

For a long time, it only triggered when you took all sd-cards out and re-booted.

I recently made a tweak to hyppo (merged into 'development' core branch) so that in addition to the above case, it would also bail out to open roms if no MEGA65.ROM was found:
- https://github.com/MEGA65/mega65-core/commit/ea3508ea50298f879ed1a5ae527ab3abd2545251

Alas, this behaviour only works in PAL, due to NTSC version choking on some raster issue that FeralChild mentioned in this issue he raised on the core:
- https://github.com/MEGA65/mega65-core/issues/362

So for ntsc, open rom usually ends up stalled on a blue-screen (which I think you can break out of with RUNSTOP-RESTORE to get to the `READY.` prompt if needed for now).

dansanderson replied:
It sounds like I would need the latest dev core in slot 0 for this to work, yeah? On my machine the boot process stops with an error message, and I have 683-cartflash in slot 0. I'll see if I can get this behavior working and figure out where it should be mentioned.

dansanderson commented on configuring.tex:
```diff
@@ -1,424 +1,185 @@
-allow you to use the MEGA65's integrated Freezer. More details are provided in the
-'Floppy Disks, D81 Images, and the Freezer' chapter on page \pageref{cha:freezer}.
-
-\subsection{ROM File}
-
-\textbf{Original C65 ROMs}
-
-You may want to source your own C65 ROM via other means.
-There were many different versions created during the development of the Commodore 65,
-and the MEGA65 can use any of them.  However, they will not support the advanced
-features of the MEGA65, and are incomplete and buggy, as development on them ceased
-due to Commodore abandoning the C65 project.
-
-\textbf{MEGA65 Closed ROMs}
-
-There are newer versions of the MEGA65 Closed ROM under development. These ROMs improve upon the original C65 ROMs and make better use of the extra hardware capabilities that the MEGA65 has over the original C65 hardware. These ROMs are available via the filehost (\url{https://files.mega65.org}), but only to owners of the MEGA65, who will need to login to the filehost with their credentials in or ...
```
The new section I just added to Upgrading about alternate ROMs covers this now.

dansanderson commented:
OK, I have addressed the comments so far. Still pending:

* Slot 0 flashing. lydon to review slot 0 flashing instructions when the feature is finished.
* Cart selection of cores. To be written once the feature is finished.
* M65Connect and Etherload. To be written once the feature is finished. The latest `mega65_ftp` is now in a good state to be used by M65Connect as a dependency for a reasonable UI wrapper.
* Soft wrap vs. hard wrap. I'll try to figure out if there are better best practices with LaTeX Workshop around this. I tend to agree that source files should be easy to edit in simple text editors and not require specific IDEs if possible, and soft wraps are less friendly to common reading experiences (but more friendly to common editing experiences).
* Consider replacing `\em` with `\underline` throughout.
* Short URL for ROM FAQ.

dansanderson commented:
FYI LaTeX Workshop uses [latexindent.pl](https://github.com/cmhughes/latexindent.pl) for auto-formatting. I'll have to test it to see if it's destructive to our custom listing environments (e.g. \screenoutput), and it'll make a huge change the first time it runs. I'd like to do this outside this PR. If it's workable, I'll install it and use it, and hopefully we'll get hard-wrapped source that's easy to manage.

gurcei commented:
PS. I also think @Impakt might not have had a chance to review your new Chapter 7 and all your changes to `appendix-basic65.tex`, as I think you made those commits after his proof-read run.

(In the case of the latter, I think he reviewed your initial basic65 changes from an earlier, but significantly more changes came in a later commit)

So just in-case, will put them on his radar in case he wants to look them over.

I'm working my way through them now too.

gurcei commented:
PS. Maybe it was a little hasty to merge this in, feel like it would've been better to wait until I had finished my chapter 7 and basic review and added my notes for it into this PR too.

Ah well, it's done now, but feel like in future, it'd be worth waiting for green ticks from the reviewers on this, especially when the changes are significant and meaty.

dansanderson commented:
I apologize if merging to master looked like I was closing off further feedback, that wasn't my intention. I was expecting that we'd continue the review on this PR. Github should allow us to post more comments here. Let me know if there are issues with that. We can also leave comments directly on the merged commit.

If it feels like I'm rushing the schedule a bit, it's because we were given a looming but unspecified deadline. 😵‍💫 Trenz Electronic needs some number of months of lead time for printing ("1-3 months" was deft's guess), and I have no idea what date they want to use as the final if our plan is still to ship batch #3 in Q4. I wanted to open the review more widely via the Digest and assumed the October Digest would be too late, and I didn't want to do that from a branch. (It would not have been a big deal to do it from a branch, it just seemed nicer to clean up the branch and encourage the wider audience to submit Issues instead of everyone piling into this PR.)

I'm very interested in your feedback on Chapter 7 and the BASIC reference editing pass. As noted above, we'll add more material on M65Connect and core slot configuration when those features have been completed. If we hit our print deadline before the features are ready, then we can remove material and direct people to short URLs linking to Filehost articles for more information.

The intro to the BASIC reference has been reworked for clarity, so that deserves some scrutiny. The changes to the reference itself are minor factual corrections and style polish. I normalized reference styles primarily based on the dominant existing styles for consistency. If it helps with the review, this describes most of the changes:

* Keyword definitions are a single sentence with the keyword as an implied subject.
* Commands are verb phrases; functions and operators are noun phrases that describe the return value; parameters are noun phrases that describe the parameter value.
* In parameter lists, trailing period only used when the definition uses multiple sentences; everything in a list consistent.
* Exceptions made for control flow keywords where the existing descriptions were more useful.
* Bitmap graphics keywords say "Bitmap graphics:" at the beginning of the description. (I remember being routinely disappointed when I first started using the BASIC reference when a useful-sounding command turned out to be limited to bitmap graphics mode. 😅)
* Use the `\index` term as the `\label` term, for the few cross-references into the BASIC reference. (I discovered that index terms can contain a `\$` but labels cannot, so a reference to `CHR\$` had to be spelled slightly differently.)
* Use LaTeX n-dashes (`--`) for value ranges.

I also resolved all of the existing doc bugs pertaining to the BASIC reference, only a few.

gurcei commented on appendix-basic65.tex:
```diff
@@ -187,30 +192,57 @@ \subsection{BASIC 65 Constants}
-and storage is claimed at their first usage. They are also initialised to zero,
-whereas string variables are initialised as an empty string "".
-
-All 104 one-letter variables are declared as {\bf fast} variables.
-26 user functions are declared as {\bf fast} functions.
```
Ok, see in your new text you've made mention of fast variables, but not fast functions.

ASIDE: perhaps tidbits like this could be placed in some 'BASIC optimisation tips' section one day.

E.g. another potential BASIC optimisation I recall was a tip that `fredrikr` gave me: to speed up your BASIC programs, any frequently called subroutines should be place either:
- after (but near to) their usage/call points
- if they must be earlier than the callers, place them at the start of your program (as he mentioned basic would search for the line number by looking at the first line of the program and onwards)

dansanderson replied:
Added a note that the fast-variable idea also applies to function names.

gurcei commented on appendix-basic65.tex:
```diff
@@ -187,30 +192,57 @@ \subsection{BASIC 65 Constants}
+before use. A variable has a default value of zero for number type variables,
+or the empty string (\screentext{""}) for string type variables.
+
+The variable types and their values are as follows.
```
I noticed that later in Bit Shifter's old text, he was referring to his table of Byte/Integer/Real/String as the 'Scalar Variable Types'

I wonder if it's worth introducing them to the user this way too, to help them 'lock-in' the sense of which variables are 'deemed' to be scalar

I.e., changing this line to "The scalar variables types..."

It might mean moving up your description of a scalar variable, but let's see what you think

dansanderson replied:
Agreed, fixed. I think it reads well to move the scalar definition to the top of the variable section.

gurcei commented on appendix-basic65.tex:
```diff
@@ -187,30 +192,57 @@ \subsection{BASIC 65 Constants}
+{\bf FOR} is a keyword. It is common to use short variable names to avoid
+these hazards.
+
+Once assigned a value, a variable can be used within an expression with other
+constants, variables, functions, and operators:
```
Surprisingly, variables can be used prior to being assigned values (they'll take on their default 0/"" value)

Ah, you mention that a little later in your text, though is it worth mentioning early here too?

dansanderson replied:
Fixed, I'll just remove the "Once assigned a value" bit.

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
 \subsection{BASIC 65 Arrays}
 \index{BASIC 65 Arrays}
 
-Each array consumes the number of elements multiplied by the item size,
```
Old text gave a multi-dimension example and provided internal details on where things were stored and how much space it'd consume

So I'm guessing you wanted to minimise surface of the api with this too

Yeah, I guess I'm ok with that, maybe these things could change in future

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
+command declares (``dimensions'') an array.
+
+An array can have multiple dimensions, each with its own index separated by a
+comma. The array must be declared with the maximum value for each dimension.
```
Bit Shifter's old text mentioned how undeclared arrays would get automatically declared with an upper boundary of 10

It's an interesting tidbit, maybe worth keeping a mention of it?

dansanderson replied:
Rewrote this paragraph to make that explicit, thanks! It's more than just interesting because it's a potential source of confusion: if you use an array variable before DIM, the array is allocated with 10 elements, but then the DIM fails because it's a re-dimensioning of the array.

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
-so an array initialised with \screentext{DIM AB(10)} consists of 11 elements and accepts indexes from
-0 to 10.
-
-String arrays are more precisely expressed as \textit{arrays of string
```
While I know sharing internal details of strings is probably overkill for new users, I have to admit, this tidbit did help me in debugging some of my own issues with strings in the past.

It's one of those tidbits that you don't know you need it until you need it. Just having the words there, letting the new user just skim over them without any great insight, but then at some later time when they're delving into string-matters more deeply, they might have a faint recollection of such a detail getting mentioned in the manual.

dansanderson replied:
This is described in POINTER, which is the right place for this for now. (Also in FREAD and FWRITE, which use POINTER in an example of reading and writing variables in a file.)

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
-string array element as the argument, returns the address of the descriptor, not the string itself.
-
-\setlength{\tabcolsep}{1mm}
-\begin{center}
-\begin{tabular}{|l|l|l|l|l|}
-\hline
-\multicolumn{2}{|c|}{\bf Type \& Item Size}  & {\bf Appended}    & {\bf Range}         & {\bf Example}  \\
-\multicolumn{2}{|c|}{}                      & {\bf Character}   &                     &                \\
-\hline
-Byte     Array &  1                         &    \screentext{\&} & 0 .. 255            & \screentext{BY\&(5,6) = 23}   \\
-Integer  Array &  2                         &    \screentext{\%} & -32768 .. 32767     & \screentext{I\%(0,10) = 5}    \\
-Real     Array &  5                         &    none            & -1E37 .. 1E37       & \screentext{XY(I\%) = 1/3}    \\
-String   Array &  3                         &    \screentext{\$} & 0 .. 255 characters & \screentext{AB\$(X) = "TEXT"} \\
-\hline
-\end{tabular}
-\end{center}
```
For the "Arrays" section, re: 2nd table of Byte/Integer/Real/String that got removed

Yeah, I don't mind the deletion of this 2nd table, it was very similar to the 1st.

Though it did have that extra 'size' column to show the number of bytes each entry in the array consumed.

Could such detail be added to the earlier table instead?

dansanderson replied:
The POINTER entry is the best place for this for now. I have added memory representation notes to that entry.

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
+character on the screen, starting from (0,0) at the top left corner.
+
+\textbf{T@\&(}column{\bf ,} row{\bf )} is the screen code of the character.
+Screen codes are not the same as PETSCII codes. Refer to a Commodore reference
+for a list of screen codes.
```
ASIDE: Hehe yes, I look forward to the day we have screen codes in our mega65-book too, I'm tired of googling for them on the web :)

Feel like I recall you had a comment somewhere in here expressing similar sentiments :)

dansanderson replied:
yeah I even filed a bug so we don't forget :)

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
-\subsection{Assignment Operator}
+An {\em operator} is a symbol or keyword that performs a function in an
+expression. It operates on one or two sub-expressions, called {\em operands}.
+The part of the expression that involves the operator is treated as the result
+of the operation: the operator {\em evaluates to} the result.
```
I had to re-read this line a few times to try get the gist of it, and I still feel a bit blurry on it. Maybe the colon in the middle of it adds to the confusion, not getting what the colon wants to contribute to it.

Not sure what to suggest on this, maybe I'd just go for "The operator and its operands evaluate to a result to be used by the expression." But maybe I've missed something in what you wanted to convey, so you might have other thoughts on this

dansanderson replied:
Good note, I attempted to simplify.

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
+equality test, and proceeds to evaluate the expression and assign the result.
+
+The operators {\bf NOT}, {\bf AND}, {\bf OR} and {\bf XOR} can be used either
+as logical operators or as boolean operators. A logical operator join two
```
> A logical operator *join* two

typo: joins

dansanderson replied:
Fixed.

gurcei commented on appendix-basic65.tex:
```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
+the order of execution. In the expression \screentext{A * (A - B) * B}, the
+subtraction is performed first.
+
+The complete set of operators and their order of predecence are summarized in
```
1. typo: precedence

2. british spelling: summarised

dansanderson replied:
Both fixed. Thanks for reading so closely! 😅

gurcei commented on appendix-basic65.tex:
```diff
@@ -672,17 +776,18 @@ \subsection{ASC}
-               the string argument and returns its numeric code value.
-               The name was apparently chosen to be a mnemonic to ASCII,
-               but the returned value is in fact the so-called PETSCII code.
+\item [Usage:] The PETSCII code of the first character of
```
Ok, I noticed for almost all functions, you've removed any prefacing verb (like 'returns', 'generates', etc)

I kinda miss having a verb like 'Returns', it feels a bit empty without it.

So I'm wondering, for any functions, should we replace '[Usage]:' with '[Returns]:'

Thanks to your recent label refinements, hopefully we could easily catch all these cases by searching for "{BASIC 65 Functions!"

dansanderson replied:
Accepted the suggestion to change "Usage" to "Returns." Nice! Also found some commands that were mis-labeled as functions. :) Also I intentionally kept "Usage" in special cases where it was more important to use behavioral language even if it's technically a function, such as FN() and USR().

In my mind, "returns" is a C-like term and not a BASIC term, and would be a special term that needs to be defined as a synonym of "evaluates to." "Evaluates to" is more accurate at the level of abstraction for BASIC functions, operators, and expressions. But that's nit picking. :)

gurcei commented on appendix-basic65.tex:
```diff
@@ -672,17 +776,18 @@ \subsection{ASC}
+\item [Usage:] The PETSCII code of the first character of
+               the string argument, as a number.
 \item [Remarks:]
                {\bf ASC} returns zero for an empty string, whose behaviour
                is different to BASIC 2, where \screentext{ASC("")} gave an error.
```
> whose behaviour is different to BASIC 2

This was existing text, but still maybe it could be a chance to change it to something like:
- "a behaviour that differs to BASIC 2,"

dansanderson replied:
I gather that "different to" is acceptable in British English, but "different from" is more universally common. I have changed all instances.

aowendev replied:
The team made the conscious decision to go with Australian English for the user guide over my preference for American English. As such, "different to" is the correct usage.

dansanderson replied:
Thanks for mentioning it. I'll change it back to "different to." It's not a big enough difference to override a previous style decision.

gurcei replied:
Oh, reflecting back on this, I think the aspect that was bugging me was more the use of the word "whose" (wasn't as touchy about "different from" vs "different to"), so my suggested re-wording was more an attempt to avoid using "whose", which sounds a little odd to my ears. "Whose" feels more natural to my ears for living things ("Mary had a little lamb, whose fleece was white as snow"), but feels out of place here.

Ok, googling now, I see examples of "whose" being used for inanimate objects too, though the examples all seem to from classy literature :D (Sure, our mega65 manual is classy in it's own way too, ofcourse ;) )

gurcei replied:
(I'll just mark this one as unresolved for now, just to assure to hear your thoughts on it)

dansanderson replied:
I fixed it: https://github.com/MEGA65/mega65-user-guide/commit/5c66d405c9ee8ab622817a0d9681f3044fe25fd8

gurcei commented on appendix-basic65.tex:
```diff
@@ -1536,7 +1654,7 @@ \subsection{CHANGE}
       \item \specialkey{RETURN} exit the command, and don't replace the current match
     \end{itemize}
-\item [Remarks:] Any un-shifted character that is not part of the string can be
+\item [Remarks:] Any punctuation character that is not part of the string can be
```
Bit shifter had previously said "Any un-shifted" character.

The older text appears more accurate, as I tried typing `FIND ZPRINTZ`, and it successfully found all PRINT calls within my code.

dansanderson replied:
Agreed that I have narrowed the description inaccurately, but "un-shifted" doesn't work. I think he meant "any character in the un-shifted character set" which I don't want to explain anywhere near here. You hold down Shift to type many of these characters, so "un-shifted" is confusing. Suggestions welcome.

gurcei replied:
Hmm, yeah, it's tricky to do it without being overly wordy eh. Maybe:

"Almost any character(footnote) can be used in place of '/')."

Then the footnote could elaborate that we mean that it is any character formed by a single keypress, without the need for holding down the SHIFT or MEGA keys.

dansanderson replied:
"Almost any" looks fine, even without needing to be a complete explanation. Another try: https://github.com/MEGA65/mega65-user-guide/commit/7b1004c3b745327babd5b093643e9834549a9963

gurcei commented on appendix-basic65.tex:
```diff
@@ -2136,9 +2257,7 @@ \subsection{COLLISION}
-\item [Info:] {\bf COLLISION} wasn't completed in BASIC 10, and
-               a working implementation will be available in a future BASIC 65 update.
-
+\item [Info:] {\bf COLLISION} wasn't completed in BASIC 10. It is available in BASIC 65.
```
My last memories of it were that it still doesn't work perfectly on our MEGA65 yet
- https://github.com/MEGA65/mega65-user-guide/issues/210

The last comment from lydon mentioned in that ticket that it has recently been fixed, so would be worth one final confirmation by running the example from the manual on the latest dev core

dansanderson replied:
I changed it because I tested the example and it worked.

gurcei commented on appendix-basic65.tex:
```diff
@@ -2166,16 +2285,26 @@ \subsection{COLOR}
+\item [Format:] {\bf COLOR} colour
+\item [Usage:] Sets the foreground text colour for subsequent {\bf PRINT} commands.
+
+               {\bf colour} the palette entry number, in the range 0 -- 255
```
I think the true range of this is still 0-31 (don't think text modes provided by BASIC can display more than 32 colours)

dansanderson replied:
Confirmed, thanks. Fixed.

gurcei commented on appendix-basic65.tex:
```diff
@@ -3470,8 +3569,10 @@ \subsection{EDMA}
 \begin{description}[leftmargin=2cm,style=nextline]
 \item [Token:] \$FE \$21
 \item [Format:] {\bf EDMA} command{\bf,} length{\bf,} source{\bf,}
-                 target [{\bf,} sub{\bf,} mod]
+                 target
```
Just wanted the backstory on the removal/hiding of those extra params bit shifter added: "target **[{\bf,} sub{\bf,} mod]**"
- have they genuinely been removed from the rom?
- or are we just hiding them due to confusion described in past discord convos, in relation to his ticket/pr?
  - https://github.com/MEGA65/mega65-user-guide/issues/518
  - https://github.com/MEGA65/mega65-user-guide/pull/517

dansanderson replied:
These arguments don't do anything yet due to gaps in the DMA. They set records $09 and $0a+$0b at the end of the 12-byte EDMA structure, but these are non-functional in the DMA right now. Of the addressing modes ($09), only Linear (the defaut) and Hold are implemented, and IIRC setting Hold via $09 doesn't work yet. Even if Hold did work via $09, you can set Hold using the upper bits of the address BANK byte. (Notably, Direction is only available on the BANK byte and is not an addressing mode. As noted in 518, you can hack this by setting those bits in the address field in the EDMA command.)

IMO, this is the wrong API for a BASIC command to begin with. I would much rather EDMA support explicit flag arguments, as a separate bit field or a string with an internal syntax. Because the current arguments do nothing, now is the time to rescind them, so we can replace them with better arguments later. Even if we do end up liking (or settling for) this API, we can reinstate the arguments when the fields are fixed in the core.

I'd rather that EDMA do something like this:
```
EDMA command, length, source, target, source_mode, target_mode, modulo_size, job_options
```
A mode argument can be a number to set flags numerically: use bits 0 and 1 to select the four addressing modes (linear, modulo, hold, XY MOD), and bits 6 and 7 to set direction and I/O register visibility. Or it could be a string value, setting those flags by letter, e.g. "LR" for linear reversed, or "M" for modulo. (I wouldn't offer bare letters in this case, so flags could be from a variable.)

I'm not sure what's most useful for exposing job options to BASIC. Maybe it's a string with special PETSCII-based syntax, or a variable argument list that just accepts option codes and parameter bytes in alternating positions. Maybe we never expose job options to BASIC.

gurcei replied:
No probs, just happy to capture the backstory of thought on this.

gurcei commented on appendix-basic65.tex:
```diff
@@ -4375,20 +4455,19 @@ \subsection{FOR}
 
 \newpage
-\label{foregroundcommand}
 \subsection{FOREGROUND}
```
Just curious on the backstory of the removal of the label?

dansanderson replied:
It is unused.

For BASIC commands with labels, I'm standardizing on having the label match the index term, instead of inventing a new syntax every time someone needs a label. I went to change this one and noticed nothing refers to it, so I removed it.

gurcei commented on appendix-basic65.tex:
```diff
@@ -4375,20 +4455,19 @@ \subsection{FOR}
-\item [Remarks:] {\bf COLOR} also has the ability to change the foreground colour.
+\item [Usage:] Sets the foreground text colour for subsequent {\bf PRINT} commands.
+
+               {\bf colour} the palette entry number, in the range 0 -- 255
```
as before with COLOR, I think the real range is 0-31

dansanderson replied:
Fixed.

gurcei commented on appendix-basic65.tex:
```diff
@@ -4375,20 +4455,19 @@ \subsection{FOR}
+               {\bf colour} the palette entry number, in the range 0 -- 255
+
+               Refer to the table under {\bf BACKGROUND} on page \pageref{colourtable} for
+               the palette entry numebrs and their corresponding colours in the default system palette.
```
typo: numebrs / numbers

dansanderson replied:
Fixed.

gurcei commented on appendix-basic65.tex:
```diff
@@ -5579,15 +5669,15 @@ \subsection{LEFT\$}
 \item [Format:] {\bf LEFT\$(}string{\bf,} n{\bf)}
-\item [Usage:] Returns a string
-               containing the first {\bf n} characters from the
+\item [Usage:] The first {\bf n} characters from the
```
Prefer something similar to old text, "**A string containing** the first...
(likewise for RIGHT$)

dansanderson replied:
Fixed in both cases.

gurcei commented on appendix-basic65.tex:
```diff
@@ -5579,15 +5669,15 @@ \subsection{LEFT\$}
+               {\bf string} a string expression
 
-               {\bf n} a numeric expression (0-255).
+               {\bf n} a numeric expression (0-255)
```
Saw you had taken the "(0 -- 255)" double-hyphen approach in other locations, so thought maybe this ought to change that way too.

dansanderson replied:
Thanks for mentioning these! I did a regex search and I believe I caught them all (avoiding the ones that are actually in code samples, of course).

gurcei commented on appendix-basic65.tex:
```diff
@@ -5699,6 +5788,59 @@ \subsection{LINE}
+
+
+\newpage
+\subsection{LINE INPUT}
```
Awesome that you documented this more thoroughly, thank you :)

gurcei commented on appendix-basic65.tex:
```diff
@@ -5751,43 +5894,39 @@ \subsection{LINE INPUT\#}
-               If the second number is omitted, the last BASIC line
-               is assumed.
+               Given a range of line numbers, {\bf LIST} lists all lines in
+               that range. A range can be two numbers separated by a hyphen
+               (\screentext{-}), a starting number and a hyphen, or a hyphen
+               and an ending number.
```
I feel like we should clarify the impact of omitting the starting range or ending range numbers.

dansanderson replied:
Took a crack at it.

gurcei commented on appendix-basic65.tex:
```diff
@@ -5751,43 +5894,39 @@ \subsection{LINE INPUT\#}
 
 \item [Remarks:]
-
                 The optional parameter {\bf P} enables page mode.
                 After listing 24 lines, the listing will stop and display
```
I tried this in 80x50 text mode of the latest beta rom and it will list 49 lines and pause there.

I'm 50/50 on whether we elaborate on this case though, maybe it's not necessary.

dansanderson replied:
Changed to "After listing a screenful of lines." :)

gurcei commented on appendix-basic65.tex:
```diff
@@ -6203,6 +6351,10 @@ \subsection{MEM}
                  stops with a \screentext{?OUT OF MEMORY ERROR}.
 
+                 Some direct mode commands like {\bf RENUMBER} use memory in
+                 banks 4 and 5 and do not honor {\bf MEM} reservations. Such
```
british spelling: honour

dansanderson replied:
Fancy! ;) Fixed.

gurcei commented on appendix-basic65.tex:
```diff
@@ -7237,49 +7401,61 @@ \subsection{PLAY}
-  consider the {\bf SOUND} command instead). Just skip the voices you're not using with
-  {\bf PLAY}, by leaving those arguments empty:
+  Similarly, the portamento directive slides between consecutive
+  notes at the speed you specify (1 -- 9). A value of 0 disables it. Note that
+  the gate-off behaviour of notes is disabled while portamento is enabled.
+  To re-enable the gate-off behavior, you must disable portamento (P0).
```
[NOTE TO SELF]:

Hmm, I'll have to revise my thoughts on the gate-off-disabling behaviour as it's no longer quite true... I recall someone in Discord pulling me up on this and wanting the current behaviour clarified a bit better. I'll try find that discord thread and recall my train of thought with it.

gurcei commented on appendix-basic65.tex:
```diff
@@ -7237,49 +7401,61 @@ \subsection{PLAY}
+  If a string ends with the {\bf L} directive, the pattern loops back to the
+  beginning of the string upon completion.
+
+  You can omit a string for a given voice, such as to allow a looping pattern
+  on that voice to continue playing, using empty arguments:
```
Feel like the description now is a little too sparse to convey the intent of the capability.

dansanderson replied:
I see what you mean. This is a mild rewrite but I think the emphasis is better: "You can omit a string for a given voice to allow an already playing pattern in that voice to continue, using empty arguments:"

gurcei commented on appendix-basic65.tex:
```diff
@@ -7358,18 +7532,19 @@ \subsection{POINTER}
                 the string descriptor, not the string itself.
-                The string descriptor consists of three bytes
-                (length, string address low, string address high).
+                The string descriptor consists of three bytes:
+                length, string address low, string address high.
```
For the info on the string descriptor, should we give an extra hint that string address low/high refer to bank1 (i think?)

dansanderson replied:
Excellent catch, added a note.

gurcei commented on appendix-basic65.tex:
```diff
@@ -8398,10 +8521,10 @@ \subsection{RND}
 \item [Token:] \$BB
 \item [Format:] {\bf RND(}type{\bf)}
-\item [Usage:] Returns a pseudo random number.
+\item [Usage:] Generates a new pseudo random number.
```
Feel like you've avoided use of a starting verb up till now, so maybe this needs to follow suit, but could also tie in with the possibility of replacing '[Usage:]' with '[Returns:]' for all functions.

dansanderson replied:
Updated. Also tidied up the explanation slightly.

gurcei commented on appendix-basic65.tex:
```diff
@@ -8436,7 +8559,7 @@ \subsection{RPALETTE}
 \item [Token:] \$CE \$0D
 \item [Format:] {\bf RPALETTE(}screen{\bf,} index{\bf,} rgb{\bf)}
-\item [Usage:]  Returns the red, green or blue value of
+\item [Usage:]  The red, green, or blue value of
```
Don't believe the ',' before the 'or' is necessary.

dansanderson replied:
per Oxford comma remark, above

gurcei commented on appendix-basic65.tex:
```diff
@@ -9218,8 +9335,7 @@ \subsection{SGN}
+\item [Usage:] The sign of a numeric expression, as a number.
 
                 \begin{itemize}
                     \item {\bf -1} negative argument.
```
> \item {\bf -0} zero.

Trying it on a recent rom, "PRINT SGN(0)" returns 0 (not -0). I haven't checked with older roms as yet.

dansanderson replied:
Looks like an old typo, there's no such thing as -0 in BASIC. Fixed.

gurcei commented on appendix-basic65.tex:
```diff
@@ -9367,8 +9485,10 @@ \subsection{SPC}
 \item [Format:] {\bf SPC(}columns{\bf)}
-\item [Usage:] Skips {\bf columns}. \\
-               The effect is similar to pressing \megakey{$\rightarrow$}
+\item [Usage:] A string of cursor-right PETSCII codes, suitable for printing
+               to advance the cursor the given number of columns.
```
Describing it this way made me curious to try if I could do the following: `A$=SPC(10)`, though that gave me a syntax error.

So it seems SPC is exclusively used by PRINT? If so, I'm wondering if this line should explicitly mention it.

dansanderson replied:
Actually an error in this entry! It's SPC$(), not SPC(). 😅 Fixed.

gurcei replied:
Oh wait, I gave `PRINT SPC$(10);"HELLO"` a try and it didn't produce any spaces though.

I took a look back at the C64 reference manual, and it was listed there as SPC() also (not SPC$), so it genuinely does seem to be SPC()! (though I agree this does seem to break a convention of string-related functions ending with $...

So feel like in your more recent commit, you'll need to revert from SPC$ back to SPC.

And I suppose this SPC() just is so unique, that doing an `A$=SPC(10)` is not permissible for some reason, and it only works with `PRINT` :(

dansanderson replied:
sorry I confused myself because A$=SPC$(10) didn't produce an error, but that's because it's accessing undimensioned array SP$ which has 10 elements. I'll fix it back. Seemed like a weird thing to be wrong. :)

gurcei commented on appendix-basic65.tex:
```diff
@@ -9746,62 +9869,12 @@ \subsection{SYS}
-
-               If the address value is 16 bit (\$0000 - \$FFFF), the bank value,
-               that is currently valid (see {\bf BANK}) is examined. A
-               bank value of 128 lets the current mapping persist. That is:
-               RAM is only available at the address range (\$0000 - \$1FFF),
-               while BASIC ROM, KERNAL and I/O occupy the rest.
-               Short machine language programs may use the address range
-               (\$1800 - \$1FFF) which is only used by BASIC while a
-               graphics screen is open.
```
Just curious why this was omitted, seems a handy reminder of the impact of BANK 128 and some advice for where to squeeze in a short assembly routine. Worthy of a new home?

Feel like I need to refresh my memory on SYS-behaviour with MAP and then re-read what you've got here. We've had some past discussions on discord about it, so that might help jog my memory.

dansanderson replied:
The updated description with regards to BANK is complete and more accurate. This section ends with a reference to the Memory chapter, which fully explains why SYS behaves the way it does.

I elided the factoid about $1800 because there's too much complex information here already. It's a useful fact but you need to know other things explained in the Memory chapter to understand why.

gurcei commented on appendix-basic65.tex:
```diff
@@ -10630,6 +10751,14 @@ \subsection{VSYNC}
 
                {\bf raster line} (0 - 311) for PAL, (0 - 262) for NTSC mode.
```
did we want the '--' hyphens here too?

gurcei commented on appendix-basic65.tex:
```diff
@@ -10823,7 +10956,7 @@ \subsection{WPOKE}
 \item [Token:] \$FE \$1D
 \item [Format:] {\bf WPOKE} address{\bf,} word [{\bf,} word ...]
-\item [Usage:]  Writes one or more words into memory
+\item [Usage:]  Writes one or more 16-bit words into memory
                 or memory mapped I/O, starting at {\bf address}.
```
Is it worth mentioning that the MEGA65's POKE/WPOKE differ slightly to the C64's POKE,  in that they don't produce an "?ILLEGAL QUANTITY ERROR" if the poked value is out of range (e.g., higher than 255 for poke, higher than 65535 for wpoke). This was a capability Bit Shifter added.

Bit shifter cropped/AND'ed the value with some max value mask (e.g. for poke, a value of 256=0, 257=1)

dansanderson replied:
This is already mentioned in POKE: "If {\bf value} is in the range of 0 -- 255, this is poked into memory, otherwise the low byte of value is used." (Not quite how I would have phrased it, but it's mentioned. :) ) I added a similar fact here for completeness.

gurcei replied:
Ah nice, though now that you've directed my eyeballs to the POKE command's text, I noticed it wrote there:

> because POKE uses the lowy byte anyway.

Was this 'lowy' a typo of 'low'?

gurcei reviewed (commented):
Ok, here's my review on Chapter 7 and the BASIC 65 reference chapter :)

Consisting of 44 comments (just in-case github auto-hides some again :D)

dansanderson commented on transferring-files.tex:
```diff
@@ -0,0 +1,173 @@
+
+The command-line tools are in a sub-folder named ``M65Connect Resources,'' such as: {\tt M65Connect Resources/mega65\_ftp}
+
+\section{Enabling Network Listening}
```
Naturally. Etherload and cartflash material are considered unfinished until those products are feature-locked for release and the content reviewed by their authors. 👍

dansanderson commented on transferring-files.tex:
```diff
@@ -0,0 +1,173 @@
+
+You can copy D81 virtual disk images to your MEGA65-formatted SD card using any PC, without any special tools or software. Your PC will recognize the data region of the SD card as a FAT32 partition. If you use this method, be aware that some PC operating systems may have unwanted side effects, such as fragmentation of SD card files, or extraneous files created by macOS Finder. These effects are h ...
+
+The fastest and most reliable way to transfer files between your PC and your MEGA65 is with an Ethernet cable.\index{Networking!Ethernet} You connect one end of the cable to the RJ45 jack on the rear of the MEGA65. You can connect the other end to your local area network (LAN) router or switch, or connect it directly to your PC. You use software on your PC to initiate file transfers, in either di ...
```
it's possible with a standard ethernet cable. There's a technological standard that we've implemented where two direct connected hosts negotiate the TX/RX pair in software.

dansanderson commented on transferring-files.tex:
```diff
@@ -0,0 +1,173 @@
+\hline
+{\tt mkdir {\it dirname}} & Create a sub-directory on the MEGA65 SD card. \\
+\hline
+{\tt cd {\it dirname}} & Change the current working directory on the MEGA65 SD card. \\
```
Done.

dansanderson commented on appendix-basic65.tex:
```diff
@@ -1,8 +1,10 @@
 
-\section{Commands, Functions and Operators}
+
+\section{Commands, Functions, and Operators}
```
It's called the Oxford comma and it's a matter of style. It is used in lists of three or more items. I use it because there are cases where it is needed to resolve ambiguity, and it's important to be consistent. This specific case is not necessarily ambiguous, but if there's a chance that it reads as two things and not three, it trips up the reader slightly trying to parse it, and I prefer to just always use it to keep the reader flowing forward.

https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/

gurcei replied:
Ok, after reading that, I see where it's coming from now, though it still feels a little cringey and foreign to my eyes (a bit like the commas/full-stops before end-quote matter too). Ah well, will leave it be for now, maybe it'll grow on me. Though if it pops on the radar of other reviewers, maybe we can ponder it again.

dansanderson commented on transferring-files.tex:
```diff
@@ -0,0 +1,173 @@
+% mega65_ftp -i 192.168.0.65 -c 'put mydisk.d81' -c 'exit'
+\end{verbatim}
+
+The tool will start a session, execute the commands, then terminate. Be sure to issue the {\tt exit} command as the final command to reset the MEGA65, or reset the MEGA65 manually after the file transfer has completed.
```
Yes, this section is unfinished until M65Connect has implemented the new changes. It will tour the M65Connect UI, and recommend it as the primary way people transfer files.

dansanderson commented on appendix-basic65.tex:
```diff
@@ -187,30 +192,57 @@ \subsection{BASIC 65 Constants}
 \subsection{BASIC 65 Variables}
 \index{BASIC 65 Variables}
 
-Each scalar variable consumes 8 bytes of storage in memory.
-The reserved area in bank 0 from \$F700-\$FEFF can store 256 variables.
```
Yes, this will go in a chapter on BASIC memory use.

There is an argument that this belongs in a description of the BASIC API because there is a (new) BASIC keyword that returns the memory location of a variable, so describing the data structure that it points to is necessary to make that keyword useful. But this is an introductory (and cursory) tour of BASIC features, so it doesn't belong here.

dansanderson commented on appendix-basic65.tex:
```diff
@@ -2166,16 +2285,26 @@ \subsection{COLOR}
+               {\bf colour} the palette entry number, in the range 0 -- 255
+
+               Refer to the table under {\bf BACKGROUND} on page \pageref{colourtable} for
+               the palette entry numebrs and their corresponding colours in the default system palette.
```
Fixed.

dansanderson commented on appendix-basic65.tex:
```diff
@@ -4165,14 +4263,18 @@ \subsection{FIND}
                 \specialkey{NO\\SCROLL} can be used to pause the output.
 
-\item [Remarks:] Any un-shifted character that is not part of
+\item [Remarks:] Any punctuation character that is not part of
                  the string can be used instead of the slash {\bf /}.
```
(See reply above.)

dansanderson commented:
@gurcei Thanks again for the detailed review! I have incorporated and addressed all of these.

"un-shifted" vs. "punctuation" characters deserves another look, I agree that my attempted improvement could go further but I don't think "un-shifted" is appropriate. Let me know if you think of a better term or explanation.

I'm interested in your reaction to my EDMA change explanation.

https://github.com/MEGA65/mega65-user-guide/commit/23c74b931ac5820e563c5401c6baa0730447876a
